rfcs/rfc-external-project-and-gem-docs.md

```diff
@@ -0,0 +1,54 @@
+# Establish process for external project and Gem documentation
+
+## Summary
+
+This RFC proposes to clarify whether documentation about external projects and Gems can be included in O3DE Documentation (`o3de.org/docs`). For the context of this RFC, "external" refers to those that live outside of the `o3de` repository.
+
+External projects and Gems commonly fall under the following scenarios:
+- They live outside of the `o3de` organization, such as those created by O3DE partners.
```

Comment:
I'm not seeing the same confusion from the wording here, although I did propose some changes for clarity, to help remove ambiguity. "External" has been defined here as anything outside of the o3de engine repo (i.e. o3de/o3de). Two examples were then presented: 1) an O3DE object that lives outside of the o3de GitHub organization; and 2) an O3DE object that is inside the o3de GitHub org, but not in the engine (e.g. o3de/o3de-multiplayersample and o3de/o3de-extras). The distinction between 1) and 2) becomes important later on, in the suggestion section, where it is suggested that objects that fall under category 2) can be considered for documentation within o3de.org. Does that clear things up? Or do you still see confusion here?

Reply:
Maybe I am confused then... If we used more precise terminology here it would make things more clear. We should not define "External" as it should not be a consideration for documentation. We should define "Canonical vs. Non-Canonical" and further in/out of org as a decision point. Canonical is by definition in org and should have documentation, Non-Canonical that is in org may also have docs but not required, Non-Canonical out of org should not have any documentation though we may refer to those repos/objects in the documentation. I see no benefit to banning in org non-canonical repos/objects documentation from the web site. We should also avoid saying "not in engine" and use the "Core vs Non-Core" when we refer to canonical repo/objects if we actually mean in the not in the core engine (o3de/o3de) repo, as all canonical repos like o3de-extras are part of the engine, I would argue ALL o3de repos are part of the engine...
Is there some sort of practical limitation I am unaware of on what we document on the web site, besides effort? I understand there are a lot of objects but I don't see the reason to draw a line and say we will not document Non-Canonical in org repos/objects on the web site.

Reply:
Hey @byrcolin, thanks for bringing up this confusion and introducing more appropriate terminology.

I can replace "external" with "non-canonical" in this RFC. Other than that, it seems like we share an understanding of what documentation must be included (canonical), may be included (in-org, non-canonical), and must not be included (out-of-org, non-canonical). 

One of the examples that was ambiguous to us is PopcornFX. Though it is an official Gem for O3DE, I'm inclined to think it's non-canonical and out-of-org. Hence why it's documentation is and should be hosted by the owner. What are your thoughts?

> Is there some sort of practical limitation I am unaware of on what we document on the web site, besides effort? I understand there are a lot of objects but I don't see the reason to draw a line and say we will not document Non-Canonical in org repos/objects on the web site.

Clarifying this boundary also motivates us to ensure that the Gems and projects that we choose to maintain in O3DE Documentation are also technically supported in some way. This is important to avoid staleness of docs. For example, it doesn't make sense to have official docs on one-time game jam projects, but it would for MSP. 

The amount of significance of non-canonical in-org projects and Gems may be along a wide spectrum. That's why we propose that for non-canonical in-org repos/objects, we _can consider_ containing their docs in O3DE Documentation, though we don't guarantee it.

Reply:
Agreed.
As to if PopcornFX is canonical: I don't currently see PopcornFX gem in any canonical repo, so it is not a canonical object. However, the AWS gems are in the core engine repo, which makes them canonical even though they should not be. Many objects currently carry the canonical status because it was decided to launch with them included in the core engine repo as the object system was not fully online yet. That has changed and it is fully online now and we are discussing which objects that currently are canonical core engine objects are going to move out of the core engine repo and where they will end up. Most will end up in the o3de-extras repo, which is another canonical repo, so they will not lose the canonical status, this is just to slim down the core repo, but others like AWS gems will lose canonical status when we move them to an AWS owned repo where they belong.

Reply:
That makes sense. The first step rn is to gain better clarity about what should live in the official docs. As features move out of the engine and become non-canonical, we can adjust the documentation to reflect what's appropriate.